### ISSUE TITLE
plaidml op.pool does not handle NaN

### ISSUE THREAD

**Issue by chrisfan100** [plaidml/plaidml#313]
Hi,
Plaidml op.pool seems can't handle NaN correctly.
cudnnSetPooling2dDescriptor use parameter cudnnNanPropagation_t to clarify the behavior of NaN.
Can plaidml support NaN like cudnn ?

Thanks,
Chris

**Comment by chrisfan100** [comment]
The following is an python example which show that plaidml op.pool get wrong result when meet NaN for max pooling:

import numpy as np
import plaidml
import plaidml.keras
plaidml.keras.install_backend()
import keras.backend as K
if __name__ == '__main__':
    arr = np.array([1, 2, 3, np.nan], dtype=float)
    arr = np.reshape(arr,[1, 2, 2, 1])
    I = K.variable(arr)
    max_val = K.op.pool(I, kernel_shape=(2,2), strides=(1,1), padding=K.op.AutoPadding.VALID, data_format=K.op.PoolDataFormat.NXC, mode=K.op.PoolMode.MAX)
    result=max_val.eval()
    print("result:", result)

the output is:
result: [[[[0.]]]]

**Comment by brianretford** [comment]
We believe this might be backend specific, fundamentally we haven't investigated this thoroughly.
